Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -25,20 +25,20 @@ namespace Microsoft.MixedReality.Toolkit.Core.Services
     /// The Profile can be swapped out at any time to meet the needs of your project.
     /// </summary>
     [DisallowMultipleComponent]
-    public class MixedRealityToolkit : MonoBehaviour
+    public class MixedRealityToolkit : MonoBehaviour, IMixedRealityServiceRegistrar
```

Comment:
This change makes it possible to have a different class implement IMixedRealityServiceRegistrar. However, it doesn't seem to fix the fundamental problem that a MixedRealityToolkit item needs to exist. Is the point of this review to better abstract service registration? Or are we actually wanting to enable an ability to setup a custom registrar. I would say this review addresses the former. But imo, without doing the deep scrub to remove MixedRealityToolkit static calls for instance, we're still setting up a weird platform architecture.

Reply:
Until Instance is of type IMixedReailtyServiceRegistrar compared to MixedRealityToolkit, I don't think we've actually achieved a componentization story.

Reply:
This is the first of several PRs. The goal of this one is to be as seamless as possible for current code. The next one will pass an instance of IMixedRealityServiceRegistrar to the service instances to eliminate the need to call MixedRealityToolkit.Instance explicitly

Reply:
The complete set of changes for #3545 exceeds the guidelines for small, targeted PRs :)

Reply:
If thats a second review, I can sign off on this first wave of changes. But if this isn't planned work, Im not sure what we are intending to enable through this interface definition.

Reply:
Yes, this is definitely planned work.

The intent here is to get the interface and the first implementation reviewed. I should have the second one (pass the interface to services and some additional static->instance changes) 
prepared today.

Reply:
Long term i think we should remove this instance call.